### PR TITLE
Add updated arch diagrams

### LINKED ISSUE #2953: Add architecture diagrams
As a developer, I'd like to add an architecture diagram to highlight SCDF, Skipper, and the other moving parts.

### PR DESCRIPTION
**Pull request by markpollack**
Fixes #2953

### CODE REVIEW

File: spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc [hunk 1]
```diff
@@ -13,8 +13,8 @@ endif::omit-tasks-docs[]
 ifndef::omit-tasks-docs[]
 Applications come in two flavors:
 
-* Long-lived Stream applications where an unbounded amount of data is consumed or produced through messaging middleware.
-* Short-lived Task applications that process a finite set of data and then terminate.
+* Long-lived applications.  These applications can be part of a `Stream` in which case an unbounded amount of data is consumed or produced through messaging middleware.  Alternatively, they can be arbitrary applications that are long lived and do not necessarily use messaging middleware.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc]: Nice rewrite!

File: spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc [hunk 2]
```diff
@@ -56,19 +54,53 @@ The Data Flow server is also responsible for:
...
+As an example, the stream DSL to describe the flow of data from an HTTP source to an Apache Cassandra sink would be written using a Unix pipes and filter syntax " `http | cassandra` ".  Each name in the DSL is mapped to an application that can that Maven or Docker repositories.  You can also register an application to an `http` location.  Many source, processor, and sink applications for common use cases (such as JDBC, HDFS, HTTP, and router) are provided by the Spring Cloud Data Flow team.  The pipe symbol represents the communication between the two applications through messaging middleware.  The messaging middleware that you can use is based on what the https://cloud.spring.io/spring-cloud-stream/#binder-implementations[Spring Cloud Stream project] supports, namely:
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc]: One of the statements in-between could be:

> The Spring Cloud Data Flow team provides many sources, processor, and sink applications for common use cases (such as JDBC, HDFS, HTTP, and router)

File: spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc [hunk 3]
```diff
@@ -56,19 +54,53 @@ The Data Flow server is also responsible for:
...
+
+* https://github.com/spring-cloud/spring-cloud-stream-binder-rabbit[RabbitMQ]
+* https://github.com/spring-cloud/spring-cloud-stream-binder-kafka[Apache Kafka]
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc]: Maybe also add Kafka Streams:

>  https://github.com/spring-cloud/spring-cloud-stream-binder-kafka[Apache Kafka / Kafka Streams]

File: spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc [hunk 4]
```diff
@@ -56,19 +54,53 @@ The Data Flow server is also responsible for:
...
+* https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis[Amazon Kinesis]
+* https://github.com/spring-cloud/spring-cloud-gcp/tree/master/spring-cloud-gcp-pubsub-stream-binder[Google PubSub (partner maintained)]
+* https://github.com/SolaceProducts/spring-cloud-stream-binder-solace[Solace PubSub+ (partner maintained)]
+* https://github.com/Microsoft/spring-cloud-azure/tree/master/spring-cloud-azure-eventhub-stream-binder[Azure Event Hubs (partner maintained)]
+
+<<applications,Pre-built Stream applications>> are available for RabbitMQ and Apache Kafka.
+You can easily modify these pre-built application using a version of https://start-scs.cfapps.io/[Spring Intializr for Stream applications].
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc]: Maybe we could remove "a version of " from the statement.

File: spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc [hunk 5]
```diff
@@ -56,19 +54,53 @@ The Data Flow server is also responsible for:
...
+Data that is posted to the HTTP application will then be stored in Cassandra.
+The https://docs.spring.io/spring-cloud-dataflow-samples/docs/current/reference/html/_streaming.html#spring-cloud-data-flow-samples-http-cassandra-overview[Samples Repository] shows this use case in full detail.
+
+The architecture for the Application DSL is similar.
+
+
+.The Spring Cloud Data Flow Application DSL Architecure
+image::{dataflow-asciidoc}/images/arch-app-dsl.png[The Spring Cloud Data Flow Application DSL Architecture, scaledwidth="60%"]
+
+In this architecture, the Application DSL description that describes a group of related applications is POSTed to the Data Flow Server.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc]: Maybe remove "description" from "the Application DSL description that describes"

File: spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc [hunk 6]
```diff
@@ -56,19 +54,53 @@ The Data Flow server is also responsible for:
...
+Instead of a pipe symbol separating the applications, a double pipe symbol is used.
+As compared to the case of a single pipe, when a double pipe symbol is used the Data Flow server does not configure the applications to connect to each other over a single messaging middleware destination.
+Instead, deployment properties (not shown) set by the user are used to configure applications to have multiple producing or consuming messaging middleware destinations.
+In this example, the `baristaApp` and sends messages on either the `hotDrinks` destination or the `coldDrinks` destination.
```
Comment (sabbyanandan) [inline spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc]: Remove "and" from "the `baristaApp` and sends message .."

**Comment by markpollack** [comment]
addressed review comments and merged as 67fb3fb895f54231d9d50bdaf8e94d6a05a32e8e

**Pull request closed without merging**